Adding the helm charts for SSO application

pvmj opened the pull request:
Pushing the helm charts for SSO through my fork as I do not have permission directly to push to the master. Please review and merge into master.

lafonj commented on cachengo-sso/templates/deployment.yaml.bak:
```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1beta2
```
Did you intend to check-in this file?

pvmj replied:
deployment.yaml.bak is not needed. Removed that file.

pvmj commented on cachengo-sso/templates/service.yaml:
```diff
@@ -0,0 +1,20 @@
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
+      nodePort: 5000
```
Changed the nodePort from 5000 to 31001. Also, as suggested reading this value from values.yaml.

pvmj commented on cachengo-sso/templates/deployment.yaml:
```diff
@@ -0,0 +1,56 @@
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+          - name: NN_SEARCH_ADDRESS
+            value: "nn_search:1323"
```
In the values.yaml, created a section named envs and added variables there. In the deployment.yaml, reading the values from values.yaml.

pvmj commented on cachengo-sso/templates/deployment.yaml:
```diff
@@ -0,0 +1,56 @@
+          - name: NN_SEARCH_ADDRESS
+            value: "nn_search:1323"
+          - name: FACEREC_ADDRESS
+            value: "facerec:5000"
```
In the values.yaml, created a section named envs and added variables there. In the deployment.yaml, reading the values from values.yaml.

pvmj commented:
Hi Jimmy,

I have updated the charts with the below.

1. Created new charts without cachengo as part of the chart names. 
2. Removed the cachengo-sso chart and added the new charts and committed them.
3. In today's commit, I added the charts for facerec & nnsearch also. But, as you know, for deploying sso chart, we do not need them now, as they are already inside the charts folder (tar format) of sso chart.
4. When I tested the charts installation, I could see the db volumes mapping was successful. I placed some files on the docker host and was able to view them inside the container.
5. I did not understand from your comment - "I don't see any persistent volume claims created after chart installation"? Even in the docker containers creation commands that you placed in the forum, I did not see anything specific to that. Please let me know if I miss anything and I will fix it.
6. I have committed the updated charts. Please review and merge if no issues.

Thanks & Regards,
PVMJ

lafonj commented:
Hi Manoj,
I don't see the changes reflected here in the pull request. Are you sure you pushed them? 

About the persistent volume: when you run `kubectl get pvc` do you see the persistent volumes for nn_search and sso? I believe you are missing something like: https://github.com/mdn/helm-charts/blob/master/mysql/templates/pvc.yaml

pvmj commented:
Hi Jimmy,

I did commit the new changes. I am not able to connect to my system now. I will cross check tomorrow morning the first thing. In the meanwhile I have sent the charts to your email. Can you please verify them? I have incorporated all the changes, except the PVC verification. I will do that tomorrow.

Thanks & Regards,
Manoj

pvmj commented:
> Hi Manoj,
> I don't see the changes reflected here in the pull request. Are you sure you pushed them?
> 
> About the persistent volume: when you run `kubectl get pvc` do you see the persistent volumes for nn_search and sso? I believe you are missing something like: https://github.com/mdn/helm-charts/blob/master/mysql/templates/pvc.yaml

Hi Jimmy,

My bad. I committed the charts, but missed pushing them. I did that in the morning. Please check and let me know.

Thanks & Regards,
PVMJ

pvmj commented on facerec/templates/deployment.yaml:
```diff
@@ -0,0 +1,43 @@
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 80
```
Done. Now reading the value of 5000 from values.yaml

pvmj commented on nnsearch/templates/deployment.yaml:
```diff
@@ -0,0 +1,54 @@
+            value: "{{ .Values.envs.var1Value}}"
+          ports:
+            - name: http
+              containerPort: 80
```
Done. Now reading the value of 1323 from values.yaml

pvmj commented on facerec/templates/deployment.yaml:
```diff
@@ -0,0 +1,43 @@
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
```
Done. Added the chart name here

pvmj commented on nnsearch/templates/deployment.yaml:
```diff
@@ -0,0 +1,54 @@
+          - name: "{{ .Values.envs.var1}}"
+            value: "{{ .Values.envs.var1Value}}"
+          ports:
+            - name: http
```
Done. Added the chart name here

pvmj commented on sso/values.yaml:
```diff
@@ -0,0 +1,53 @@
+
+envs:
+  var1: NN_SEARCH_ADDRESS
+  var1Value: nn_search:1323
```
Done. Appending the release name to the env variable.

pvmj commented on sso/values.yaml:
```diff
@@ -0,0 +1,53 @@
+  var1: NN_SEARCH_ADDRESS
+  var1Value: nn_search:1323
+  var2: FACEREC_ADDRESS
+  var2Value: facerec:5000
```
Done. Appending the release name to the env variable.

pvmj commented on sso/templates/deployment.yaml:
```diff
@@ -0,0 +1,56 @@
+          - mountPath: /db
+            name: sso-db-volume
+      volumes:
+      - name: sso-db-volume
```
Created persistent volumes and claims for both sso and nnsearch.

pvmj commented on nnsearch/templates/deployment.yaml:
```diff
@@ -0,0 +1,54 @@
+          - mountPath: /db
+            name: nn-db-volume
+      volumes:
+      - name: nn-db-volume
```
Created persistent volumes and claims for both sso and nnsearch.

pvmj commented:
Hi Jimmy,

I could fix the PVC issue that I was reporting yesterday. Pushed the changes with that fix. 

Now, I guess I fixed all the issues reported by you and the enhancement. Please review and let me know your feedback.

Thanks & Regards,
PVMJ

lafonj commented on facerec/templates/service.yaml:
```diff
@@ -0,0 +1,19 @@
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
```
This should be set to 5000 and be taken from values.yaml

pvmj replied:
Done.

lafonj commented on nnsearch/templates/service.yaml:
```diff
@@ -0,0 +1,19 @@
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
```
This should be 1323 and be taken from values.yaml

pvmj replied:
Done

lafonj commented on sso/values.yaml:
```diff
@@ -0,0 +1,53 @@
+
+image:
+  repository: cachengo/sso_example
+  tag: 1.1
```
Can we update this to 1.2?

pvmj replied:
Done

pvmj commented:
Hi Jimmy,

Thank you for your review.  I updated the proposed changes and pushed them.

Best Regards,
PVMJ

lafonj commented:
Works great. Can you squash the commits before I merge it?

pvmj commented:
Great !  Thanks !

- PVMJ

lafonj commented:
Hi @pvmj, just checking in to see if you were able to squash the commits and push again?
